rs/registry/helpers/src/firewall.rs

```diff
@@ -85,6 +94,56 @@ impl<T: RegistryClient + ?Sized> FirewallRegistry for T {
             .collect();
         Ok(Some(result))
     }
+
+    /// Get the IP addresses of all system subnet nodes in the registry, for endpoints used for core protocol services (p2p, xnet, api)
+    fn get_system_subnet_nodes_ip_addresses(
+        &self,
+        version: RegistryVersion,
+    ) -> RegistryClientResult<Vec<IpAddr>> {
+        let system_subnet_node_ids: Vec<NodeId> = self
+            .get_system_subnet_ids(version)?
+            .map(|system_subnet_ids| {
+                system_subnet_ids
+                    .into_iter()
+                    .filter_map(
+                        |subnet_id| match self.get_node_ids_on_subnet(subnet_id, version) {
+                            Ok(Some(node_ids)) => Some(node_ids),
```

Comment:
optional nit: you could return node_ids without Some(), use vec![] in the other 2 branches below and replace filter_map with flat_map, then you don't need to flatten the iterator.

Reply:
Great suggestion! Thanks!